Documentation/spec/docfx_document_schema.md

```diff
@@ -0,0 +1,197 @@
+
+For example, to add a schema for a document type `ManagedReference`, a schema file should added like `{templateFolder}\schemas\ManagedReference.json`.
+
+Schema describes the object model. The format of the object model is not constrained. YAML and JSON should both be supported.
+
+## 4. General Considerations
+* This schema will reuse keywords in JSON schema, but will not change the defition of the existing one.
+* This schema can validate and manipulate single elements in place, but will not reconstruct the object.
+
+## 5. Validation keywords
+
+### 5.1 type
+
+Same as in JSON schema: http://json-schema.org/latest/json-schema-validation.html#rfc.section.6.25
+
+> The value of this keyword MUST be either a string or an array. If it is an array, elements of the array MUST be strings and MUST be unique.
+>
+> String values MUST be one of the six primitive types ("null", "boolean", "object", "array", "number", or "string"), or "integer" which matches any number with a zero fractional part.
+>
+> An instance validates if and only if the instance is in any of the sets listed for this keyword.
+
+
+### 5.2 properties
+
+Same as in JSON schema: http://json-schema.org/latest/json-schema-validation.html#rfc.section.6.18
+
+> The value of "properties" MUST be an object. Each value of this object MUST be a valid JSON Schema.
+>
+> This keyword determines how child instances validate for objects, and does not directly validate the immediate instance itself.
+>
+> Validation succeeds if, for each name that appears in both the instance and as a name within this keyword's value, the child instance for that name successfully validates against the corresponding schema.
+>
+> Omitting this keyword has the same behavior as an empty object.
+
+### 5.3 items
+
+Same as in JSON schema: http://json-schema.org/latest/json-schema-validation.html#rfc.section.6.9
+
+> The value of "items" MUST be either a valid JSON Schema or an array of valid JSON Schemas.
+>
+> This keyword determines how child instances validate for arrays, and does not directly validate the immediate instance itself.
+>
+> If "items" is a schema, validation succeeds if all elements in the array successfully validate against that schema.
+>
+> If "items" is an array of schemas, validation succeeds if each element of the instance validates against the schema at the same position, if any.
+>
+> Omitting this keyword has the same behavior as an empty schema.
+
+## 6. Manipulation keywords
+
+### 6.1 behavior
+The value of "behavior" MUST be a string. It defines how DocFX will manipulate the property and affect the build context.
+
+The following values are allowed:
+* `uid`: If the instance is a string, it will be marked as a UID (stands for unique identifier) in the current model, and add it to the document build context.
```

Comment:
[http://json-schema.org/latest/json-schema-core.html#rfc.section.4.2](http://json-schema.org/latest/json-schema-core.html#rfc.section.4.2) In JSON schema, it usually uses `instance`.

---
In reply to: [122872017](https://github.com/dotnet/docfx/pull/1771#discussion_r122872017) [](ancestors = 122872017)